### ISSUE TITLE
Calculate missing slice locations

### ISSUE THREAD

**Issue by mlavik1** [mlavik1/UnityVolumeRendering#26]
See: https://github.com/mlavik1/UnityVolumeRendering/issues/25#issue-716760346

Apparently it's common for some DICOM files to lack the "slice location" tag. However, it should still be possible to calculate that (see thread).

**Comment by MusoAga** [comment]
Hey @mlavik1 the Slice Location is an optinal tag (https://dicom.innolitics.com/ciods/ct-image/image-plane/00201041).

You should use the Image Position Patient to calculate the dimensions, the spaces between the slices and their location. (https://dicom.innolitics.com/ciods/ct-image/image-plane/00200032). This tag is required. So you can use this saftley.

Something like:

`float space = Vector3.Distance(data[2].ImagePosition, data[1].ImagePosition);`

and 

``` 
float xDimension = resolutionX * PixelSpacing.x;
float yDimension = resolutionY * PixelSpacing.y;
float zDimension = sliceCount * space;
```

**Comment by mlavik1** [comment]
Hello again @MusoAga ! 👋 
Thank you very much!  This makes the calculations a lot easier :) 

So if I understand this correctly, the "ImagePosition" is basically the position of the slice (it's upper left corner). I suppose for 3D volume data all the "ImagePosition" values of all the slices will usually be linearly alligned (all the points make a straight line), right? 

So if I want to sort the slices by position, I would first calculate the vector between the "ImagePosition" of two arbitrary slices, and then sort each slice depending on where it is on this line. (I think! 😁)

**Comment by MusoAga** [comment]
Yes, exactly. That should do the trick :D If I am right, the ImagePosition is directly generated by the medical device. Other tags like SliceLocation, SpaceBetweenSlices are more settings set by human. So they are not accurate.  

Look here:
https://stackoverflow.com/questions/37730772/get-distance-between-slices-in-dicom
https://stackoverflow.com/questions/14930222/how-to-calculate-space-between-dicom-slices-for-mpr

Hope this helps 😄

**Comment by mlavik1** [comment]
Yes, it really helped me!
I had no idea about this, so thanks for pointing it out!

I'll see if I get time to start working on this soon.
Thanks a lot for the help 😄

**Comment by mlavik1** [comment]
Fixed in #38
